Gallery

WAEliasmith opened the pull request:
# Description
Currently this uses the dummy data with a similar chakraUI setup as the other one, putting the links to the gallery and secondary gallery on the new navbar and the new pages and dynamic routing in the router. There are still issues with the images in the events and gallery pages


# How to test

1. check out locally, and go to http://localhost:3000
2. check out the new nav links

# Screenshots/Screen Recording of Implementation
Visuals are WIP

![image](https://github.com/keypom/keypom-frontend/assets/3915233/fcfe67c5-20fb-4888-addd-c3f988184f56)
![image](https://github.com/keypom/keypom-frontend/assets/3915233/b9315b2d-334f-43bc-b69a-19b2587740a8)

WAEliasmith committed:
Added WIP Gallery, Secondary Gallery, dynamic Events page

Similar to the FYDPFrontend git repository, simple changes for now

WAEliasmith committed:
eslint fix

and Image attempt

WAEliasmith committed:
Fixed up gallery UI to better match examples

- Added external link icon on gallery page and open in new tab functionality
- Css fixes
- Event page css fixes, side by side data
- "Share Event" section
- Open in google maps for the location
- Price at the buy button

WAEliasmith committed:
Fixed style checks

WAEliasmith committed:
Added gradient

like the one on home page

WAEliasmith committed:
Gallery and events page

Removed external link logo in gallery
removed secondary market page and dynamic secondary events page
added banner and other required parts to gallery and events so it looks like the figma and example
added ticket grid at the bottom of events page with new ticket component (and dummy data).
removed blue highlights from cards
turned the purchase modal into a separate component

WAEliasmith committed:
Merge branch 'feat/nft-drops' into Gallery

WAEliasmith committed:
WIP gallery revamp

Fixed bug with sorting by id or date when switching from one to the other (keep a temp variable to ignore state updating slowly)

Fixed the date and title button to match those on the drops tag, also fixed search

partially ported in pagination, but doesnt work without it all yet.
Implemented dynamic loading, using drops as dummy data.

Moved gallery grid to a smaller component with support for loading skeletons

7axel7 committed:
Many frontend fixes and changes

Verify:
- added ticket verify button and modal
- added qr code reader to verify modal
- added hoverable instructions on how to use the verify modal
- added toasts to show if your qr code is good or not

Gallery:
- added gallery hover animation
- gave entire gallery pretty border, and thin translucent border on each card
- moved title so its like drops page
- fixed bug with the gallery sorting order, toggling it now works properly (removed id sort)
- added WIP sorting using a dropdown component from the drops
- added gallery pagination
- added loading skeletons when the info isnt grabbed yet
- switched the dummy data over to drop data so there is load times
- smart loading where it just grabs a filtered page first before getting the rest
- made gallery search bar work on name and description (and look like drop search bar)
- added (nonfunctional) sort select with a bunch of price range options

Event page:
- fixed event page to work with old dummy data
- fixed error where the page crashes instead of showing unknown ticket when the number is too high
- made it so hovering over a card on the event page turns the cursor into a pointy finger and lets you click on it to open the buy modal
- added hover animation to tickets

Sell modal:
- created sell modal on gallery page as its own component
- detected a secret key in the url (hard coded)
- pops up sell modal when the secret key is correct
- sell modal goes away and secret key disappears when you close or hit submit
- sell function is called with an inputted price when you hit submit, gives toast feedback

Other:
- restyled purchase modal
- moved PurchaseModal to its own component

7axel7 committed:
Many changes for the gallery and events page prototypes

Purchase modal:
- added placeholder toast on success purchase or failure
- fixed ticket purchase modal and made default amount 1
- made it so that when you leave the purchase modal and open it again, ticket amount and email reset
- print out email and amount values on purchase, so you can use them in the event page (small refactor)

Other:
- fixed up props for subcomponents

Sorting and Filtering:
- converted each drop to a promise so that I can filter it by price
- uses each drop as a promise so that I can filter the description (as well as the title)
- fixed up price tiers so the default is "any"
- made the sort into a dropdown
- made the sort order reverse if you click an arrow (with hover cursor) or select the current thing
- added loading to the sort order, since it cant sort unless all the items are there
- made the sort functional by filtering a combination of drops and dropdata promises by date, price, and tickets. currently filters using parts of drop data
- fixed spacing on the size-changing sort button
- used a library with a good date picker
- wrote a custom css file to make it match the desired border radius and size (and font and weight etc.)
- made the start and the end dates on the datepicker work with filtering, converting the id to date for testing

WAEliasmith committed:
A bunch of changes to get the gallery and events page into a suitable state

Events page:
- Fixed events page to be like the figma, 33% from the top title, 33% from the right location and date split, smaller fonts, removed share section
- reused the component from the gallery page (so they look similar)
- made it so the nav link on the gallery page component is deactivated
- made it so the button at the bottom opens the purchase modal instead of page navigation
- made the plus minus number thingy like the figma
- made the amount number go to the purchase modal
- made the ticket amount and email reset on modal close

Gallery page:
- added custom css to round the start and end of the date chooser
- added a new component for cards in the grid which use the pretty border
- new styling for the new component like in the picture ben sent
- made it so that the card on the gallery is clickable all the way to the edge to link to the event page
- removed other sorting options, now just tickets ascending and descending
- removed ascending descending arrow to click
- fixed loading box size
- Made banner pick a random image once its loaded
- made selector for dates go light blue in the middle and darker blue at the ends
- made selector for dates start and end with circles like the gfigma
- made selector for dates have no border when selected (and unselected) (like figma)
- changed font in 4 ways for month selector to be like other things
- made numbers all grey, but selected ones white?
- removed annoying border in selector date

Other:
- made the tiling in the gallery and events pages 2/3 the size, closer to the edge, closer together
- Fixed the styling on the purchase modal to be cleaner and have simple lables and information.
- Greyed out the checkout button when there is no email
- Fixed the styling on the verification modal, and shows the result.
- Fixed toasts wording and timing
- Fixed the styling on the sell modal

WAEliasmith committed:
eslint fix 1

WAEliasmith committed:
More styling fixes and abstraction

- fixed date styling on ticket cards
- changed gallery ticket to say "go to event"
- added `<ModalContent maxH="90vh">
        <Box maxH="90vh" overflowY="auto" p="4">` to scroll when needed for verify modal
- put an absolute 0 start full height and width z index background white box on the gallery page to match figma
- added a message when nothing is found in gallery
- made gallery page size numbers pick from a new list, with the first as default
- removed bottom padding, made text and image smaller, change title smaller and nicer for ticket card
- fixed up font in verify and purchase modal and resell modal
- added smaller padding to those three modals
- changed purchase modal ticket incrementer to clicky buttons
- abstracted ticket incrementer to its own component
- made events page stack better on mobile
- added a sort and price filter to mobile gallery
- added smaller mobile date filter
- changed secret key to a `window.location.hash` parsing so I can use a hashtag instead of question mark

WAEliasmith committed:
events WIP

WAEliasmith committed:
Merge branch 'feat/my-events-page' into gallery

WAEliasmith committed:
Integration day 1

- more eslint fixes
- hide verify ticket information until scan
- changed purchase modal text spacing to be in groups of 2
- used keypomInstance.getAllEventDrops to get benjiman.testnet drops, so new data is used
- made the search button not grow if the date picker does and is centered
- made the new data work for the banner, ticketcard media, ticket count
- some semi-dummy data such as dateCreated instead of event date, and event.id instead of location TODO: ASK
- almost got the data working on the event page but ran into errors
- went through like 3 failed methods of creating a ed25519 public key from a private key, finally got an output using `ecdh = crypto.createECDH('secp256k1')`, will check if it works when I use the key (failed methods were: tweetnacl, node-forge, crypto.createPublicKey)

WAEliasmith committed:
Daily updates for gallery

- date parsing in gallery
- populated gallery ticketcards with proper date, location, and description
- popoulated event gallery body with proper date, location, image, and description
- made gallery ticketcards nav to their id on a event page
- called geteventdrop on the url of the gallery to see if it is valid, show the "no drop" element if it isnt
- get ticket information using getTicketsForEvent and display them at the bottom of the events page
- hid the ticket availability badge if there are unlimited, added sold out badge if they are sold out, put good ticket information into ticket listing
- made tickets display title, description, date, price

WAEliasmith committed:
daily changes for gallery

- split up eventid url into accountid:eventID
- made gallery send with navlink to "benjiman.testnet":eventID
- created a keypom.ts function getKeyInformation to get drop info from public key
- replaced my ecdh = crypto.createECDH('secp256k1'); key transforming with getPubFromSecret()
- used getKeyInformation with a test private key to get keyinfo
- used the keyinfo to get drop id and then used getTicketDropInformation to get drop info
- parsed the drop info to get metadata and get the real eventId, used getEventDrop on the eventId to get the drop
- parsed the info from the drop and made it display in the key selling modal so you know what you are selling
- made all this info work on the url and not a testing key, made the url and state go away on modal cancel
- restyled the name at the start of the resell modal
- use the same logic to detect secretkeys in the QR code, and then display the ticket information below in a formatted list
- removed the arrow in the datepicker

WAEliasmith committed:
Merge branch 'feat/my-events-page' into Gallery

WAEliasmith committed:
wip new data

WAEliasmith committed:
Merge branch 'fydp' into Gallery

WAEliasmith committed:
daily gallery updates friday

- Merged ben branch
- Made search bar thinner
- Used min's marketplace contract to populate the frontend with new data, switch all of the data over to the new formatting of min's marketplace contract
- reduced hover effect on ticket card
- removed outer box on gallery
- removed divider lines around search
- Decrease padding below search box
- Change price placeholder to say near
- Reset date filters button
- merged fydp into the branch
- Increased distance between ticket title and image
- Made the ticket date text smaller and greyer
- reduced padding between date and location and button and description
- gave gallery tickets max width
- Decrease ticket border radius
- Change pill border radius to full and remove padding
- Smaller pill font 2extrasmall
- pill Higher in top right
- fixed up handleGetAllTickets in event to work with new datastructure
- fixed google maps leading to no search
- Put title above event details
- Removed white border
- Decrease but make bold labels and black
- spent forever trying to give each button seperate variables to increment and decrement, but some ungodly thing is causing the page to reload information and reset it on button press
- made gallery page work for null and undefined things on the marketplace contract
- re-enabled filtering
- fixed price filtering to work with new data
- fixed name and description search to work with new data
- fixed date filtering to match enw data
- fixed sorting to work with new data
- made all of these work for null and undefined data
- fixed date displays with from and to on gallery and events page

WAEliasmith committed:
Merge branch 'fydp' into Gallery

BenKurrek committed:
custom date range picker beginning implementation

BenKurrek committed:
date range picker finished

BenKurrek committed:
working on event preview page

BenKurrek committed:
about to refactor

BenKurrek committed:
successful refactor

BenKurrek committed:
finalized step 3

BenKurrek commented on package.json:
```diff
@@ -40,10 +40,11 @@
     "react-hook-form": "^7.43.0",
     "react-qr-code": "^2.0.11",
-    "react-qr-reader": "3.0.0-beta-1",
+    "react-qr-reader": "^3.0.0-beta-1",
```
Why was this added?

BenKurrek commented on package.json:
```diff
@@ -40,10 +40,11 @@
     "mathjs": "^11.5.1",
     "near-api-js": "^1.1.0",
     "react": "^18.2.0",
+    "react-dates": "^21.8.0",
```
Remove (since we're using my date range picker)

BenKurrek commented on src/features/all-drops/components/DropManagerPagination.tsx:
```diff
@@ -25,7 +26,11 @@ export const DropManagerPagination = ({
   numPages,
   handleNextPage,
   handlePrevPage,
+  type,
```
Change this to type = 'Rows' and remove your `if (type === undefined)` check. Also change type to be optional in props.

BenKurrek commented on src/features/gallery/components/PurchaseModal.tsx:
```diff
@@ -0,0 +1,105 @@
+  const isError = email === '';
+  return (
+    <Modal isCentered closeOnOverlayClick={false} isOpen={isOpen} size={'xl'} onClose={onClose}>
+      <ModalOverlay />
```
Please apply this fix to all your modal components because it fixes the lag issue
```suggestion
            <ModalOverlay backdropFilter="blur(0px)" bg="blackAlpha.600" opacity="1" />
```

BenKurrek commented on src/features/gallery/components/SellModal.tsx:
```diff
@@ -0,0 +1,96 @@
+  const isError = input === '';
+  return (
+    <Modal isCentered closeOnOverlayClick={false} isOpen={isOpen} size={'xl'} onClose={onClose}>
+      <ModalOverlay />
```
Same as above

BenKurrek commented on src/features/gallery/components/VerifyModal.tsx:
```diff
@@ -0,0 +1,251 @@
+      return;
+    }
+
+    // test answer
```
Remove

BenKurrek commented on src/features/gallery/components/VerifyModal.tsx:
```diff
@@ -0,0 +1,251 @@
+  event: any;
+}
+
+const accountId = 'benjiman.testnet';
```
Remove hardcoded account ID

BenKurrek commented on src/features/gallery/components/cssoverrides.css:
```diff
@@ -0,0 +1,92 @@
+/* .DateRangePickerInput {
```
Remove this file

BenKurrek commented on src/pages/Event.tsx:
```diff
@@ -0,0 +1,579 @@
+        ticketInfo: meta2.ticketInfo,
+      });
+
+      setDoKeyModal(true);
```
What is `DoKeyModal`? Can you rename this

BenKurrek commented on src/pages/Event.tsx:
```diff
@@ -0,0 +1,579 @@
+
+  const loadingdata = [];
+
+  // append 10 loading cards
```
This is 3 not 10

BenKurrek commented on src/pages/Event.tsx:
```diff
@@ -0,0 +1,579 @@
+    // sell the ticket with the secret key, give toast, and sell
+    navigate('./');
+
+    const buySuccessful = Math.random();
```
Remove this logic

BenKurrek commented on src/pages/Event.tsx:
```diff
@@ -0,0 +1,579 @@
+    navigate('./');
+    setDoKeyModal(false);
+    setSellDropInfo(null);
+    const sellsuccessful = Math.random();
```
camelCase pls

BenKurrek commented on src/pages/Event.tsx:
```diff
@@ -0,0 +1,579 @@
+    const sellsuccessful = Math.random();
+    console.log('inpuit' + sellsuccessful + input);
+
+    if (sellsuccessful <= 0.5) {
```
Remove random

BenKurrek commented on src/pages/Gallery.tsx:
```diff
@@ -0,0 +1,758 @@
+
+  const handleGetAllEvents = useCallback(async () => {
+    setIsAllDropsLoading(true);
+    let eventListings = await keypomInstance.GetMarketListings({
```
You need to paginate here instead of hardcoding a 50 limit

BenKurrek commented on src/pages/Gallery.tsx:
```diff
@@ -0,0 +1,758 @@
+    let filteredDrops: ProtocolReturnedDrop[] = [];
+    while (dropsFetched < totalSupply && filteredDrops.length < selectedFilters.pageSize) {
+      const drops = await keypomInstance.GetMarketListings({
+        limit: 50,
```
I would decrease this value to ~20 at a time

BenKurrek commented on src/pages/Gallery.tsx:
```diff
@@ -0,0 +1,758 @@
+    dropData = dropData.filter((drop) => drop !== null && drop !== undefined);
+    console.log('just resolved4', dropData);
+
+    setFilteredDataItems(dropData);
```
Only set filtered items if the getAllEvents has not finished

BenKurrek commented on src/pages/Gallery.tsx:
```diff
@@ -0,0 +1,758 @@
+    gridData = gridData.map((drop) => {
+      console.log('swefdrop: ', drop);
+      let dateString = drop.dateString;
+      if(drop.dateString == undefined) {
```
You should have all this date logic in a helper function

WAEliasmith committed:
daily gallery and events page update

- Got rid of time in gallery page
- Fixed bug where the initialdrops would overwrite the all drops
- Sped up initial drops by making it take smaller chunks and removing redundant checks
- Fixed bug where art was missing and data was gone and links were undefined in initial drops for new data
- Made descriptions properly truncate when fetching instead of displaying
- Made name truncate, switch truncates to end, fixed lengths
- Lowered location to be in line with event details
- Added padding between location block and date block
- removed more console logs
- fixed button spacing on ticket cards to be at the bottom regardless of description size
- added small top spacing to gallerygrid
- removed bottom padding from gallerygrid
- changed default ticket sort to descending
- removed top padding from event page image
- Made the url check fail if it's different in verify modal
- Made each ticketcard responsible for the amount of tickets selected with the plus or minus buttons
- made ticket purchase button display amount update with amount purchased
- disabled purchase buttons on tickets that are sold out
- made selected ticket amount in the ticket cards remain to the purchase modal
- made event questions dynamically be added to the purchase modal. Made the responses to the questions stored and put into the purchaseticket function
- made the purchase modal scroll
- made the requiredfields show up in red until they are filled out, the purchase button is grey until all required filled out
- implemented a way to switch from question to question in a single click despite the blur nonsense, but it causes so much lag and makes no sense, so it is gone for now
- made a placeholder with 3 response await thingies, one for stripe, free, and near on a conditional in the purchaseticket function
- placeholder if it is purchasenear, do placeholder signandsendtransaction

BenKurrek committed:
almost finished ticket creation page

BenKurrek committed:
first pass at ticket creation tab

BenKurrek committed:
preview modal progress

WAEliasmith committed:
secondary market and purchase WIP

WAEliasmith committed:
Merge branch 'feat/create-tickets' into Gallery

BenKurrek committed:
cleared build errors

BenKurrek committed:
first pass at review details

BenKurrek committed:
fix build

BenKurrek committed:
first pass at event creation pseudo successful

BenKurrek committed:
first pass at cost breakdown

BenKurrek committed:
ticket creation almost finished

WAEliasmith committed:
Daily update, demo working

- made event page time parsing work
- put workerpayload in local storage before near call, and made function to look for near redirect and get the workerpayload out to use in the worker
- integrated with worker to send an email after recieving a near url
- parsed the url secret to figure out when it has been redirected from near
- made the verifymodal remove all data on close
- fixed the verifymodal so it no longer uses any deprecated functions and uses the new information storage method
- made the sellmodal no longer use deprecated functions and use the new information layout
- made worker called if something is bought with stripe or other
- bugfixed a ton of integration issues with min
- fixed secondarymarket to work with new data
- got all worker requests and near requests formatted correctly
- added new data like funderID to workerpayload so it is visible
- fixed up toasts to more accurately say what is happening
- use ps58, nacl, naclUtil to create a resell signature
- made a function to let people list tickets they dont own with the keypom global secretkey
- removed console.logs

BenKurrek committed:
build errors

WAEliasmith committed:
Merge branch 'feat/create-tickets' into Gallery

WAEliasmith committed:
small fix after merge

BenKurrek committed:
getting ready for demo

BenKurrek committed:
full functioning demo

BenKurrek committed:
changes to navbar and modals

BenKurrek committed:
started work on linting errors

BenKurrek committed:
began working on QOL code changes

BenKurrek committed:
changed estimator

BenKurrek committed:
reset button for date time picker, date validation in event creation, bugfixes and more

WAEliasmith committed:
stripe and free integration

also some eslint changes

BenKurrek committed:
refactor time picker

BenKurrek committed:
refactor placeholder text for date

BenKurrek committed:
added proper redirects for events managers

WAEliasmith committed:
Merge pull request #226 from keypom/chore/ticket-creation

Chore/ticket creation

BenKurrek committed:
added sell button and removed connect button from navbar

BenKurrek committed:
added no connect button on mobile

BenKurrek committed:
fixed bugs introduced with fresh builds and no caching

BenKurrek committed:
changed worker base URL

WAEliasmith committed:
added gitattributes file

WAEliasmith committed:
changed vcproj to lf in gitattributes

WAEliasmith committed:
eslint fixes

WAEliasmith committed:
fixed a bunch of type checks

WAEliasmith committed:
Merge branch 'feat/qr-page' into Gallery

WAEliasmith committed:
Merge branch 'chore/fresh-build-tests' into Gallery

WAEliasmith committed:
removed eslint errors and warnings

WAEliasmith committed:
Fixed more check-types
